hikari/impl/cache.py

```diff
@@ -134,15 +133,16 @@ def __init__(
         # found attached to cached presence activities.
         self._unknown_custom_emoji_entries = collections.FreezableDict()
         self._user_entries = collections.FreezableDict()
-        self._message_entries = collections.LimitedCapacityCacheMap(limit=settings.max_messages)
-        self._intents = intents
+        self._message_entries = collections.LimitedCapacityCacheMap(limit=self._settings.max_messages)
 
-    def _assert_has_intent(self, intents: intents_.Intents, /) -> None:
-        if self._intents ^ intents:
-            raise errors.MissingIntentError(intents) from None
+    def _is_cache_enabled_for(self, setting: str) -> bool:
+        return self._settings.enable and getattr(self._settings, setting)
 
-    def _is_intent_enabled(self, intents: intents_.Intents, /) -> bool:
-        return (self._intents & intents) == intents
+    def clear(self) -> None:
+        if not self._is_cache_enabled_for("enable"):
+            return None
+
+        self._create_cache()
```

Comment:
what are there benefits like between this and calling "clear" on each resource to stuff like how quickly memory is deallocated/how long the call takes, etc etc

Reply:
Have not tried it, will see the differences

Reply:
```py
    def clear(self) -> None:
        if not self._is_cache_enabled_for(ENABLE):
            return None

        current, peak = tracemalloc.get_traced_memory()
        print(f"Current memory usage is {current / 10**6}MB; Peak was {peak / 10**6}MB")
        print("Collecting")
        gc.set_debug(gc.DEBUG_STATS)
        self._create_cache()
        gc.collect()
        gc.set_debug(0)
        print("Done collecting")
        current, peak = tracemalloc.get_traced_memory()
        print(f"Current memory usage is {current / 10**6}MB; Peak was {peak / 10**6}MB")
```
![image](https://user-images.githubusercontent.com/29100934/103460376-fed5ab80-4d15-11eb-9a4f-badeb82120ed.png)